src/protagonist/Engine/Ingest/AssetIngester.cs

```diff
@@ -70,14 +73,22 @@ public Task<IngestResult> Ingest(LegacyIngestEvent request, CancellationToken ca
     /// <returns>Result of ingest operations</returns>
     public async Task<IngestResult> Ingest(IngestAssetRequest request, CancellationToken cancellationToken = default)
     {
+        var asset = await assetRepository.GetAsset(request.Asset.Id);
+
+        if (asset == null)
+        {
+            logger.LogError("Could not find an asset for asset id {AssetId}", request.Asset.Id);
+            return new IngestResult(asset, IngestResultStatus.Failed);
+        }
+        
         // get any matching CustomerOriginStrategy 
-        var customerOriginStrategy = await customerOriginRepository.GetCustomerOriginStrategy(request.Asset, true);
+        var customerOriginStrategy = await customerOriginRepository.GetCustomerOriginStrategy(asset, true);
             
         // set Thumbnail and ImageOptimisation policies on Asset
-        await HydrateAssetPolicies(request.Asset);
+        await HydrateAssetPolicies(asset);

```

Comment:
Does this ticket include updating what this method does? It currently loads full `ThumbnailPolicy` and `ImageOptimisationPolicy` data but those have been replaced

Reply:
I don't think it was explicitly called out for removal, but might make sense to do it here - only reason I can see not to, is if it would be covered under the full "strip out old delivery channels" work, which I don't believe it is - so will get this updated

Reply:
actually, I've just looked at this and might make more sense in either #621 or #629 as removing them breaks the `AppetiserRequestModel` from sending values such as `kdu_max`

Reply:
stuck a note on #629